api/v1alpha1/ingressnodefirewall_types.go

```diff
@@ -18,129 +18,150 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
 // IngressNodeFirewallICMPRule define ingress node firewall rule for ICMP and ICMPv6 protocols
 type IngressNodeFirewallICMPRule struct {
-	// ICMPType define ICMP Type Numbers (RFC 792).
-	// +kubebuilder:validation:Required
+	// imcpType define ICMP Type Numbers (RFC 792).
+	// if its set the type field in the ICMP /ICMP6 header will match against this configured value otherwise its ignored.
```

Comment:
`If set, this field's value matches against the ICMP / ICMP6 header.`

How do we distinguish between "set" and icmp type = 0 ?

Reply:
type of 0 probably doesn't exists I will use configured instead of set

Reply:
code 0 + type 0 = ICMP echo reply https://en.wikipedia.org/wiki/Ping_(networking_utility)#Echo_reply :-)

Reply:
my bad don't remember them had to check the RFC anyway I reworded it now so hopefully we are good now